### PR TITLE
Greehouse Fields Builder (alpha alternative version)

### PR DESCRIPTION
**Pull request by patanj101**
**Do not merge ...**

I can finalize it after your feedback ...

I will add a demographic_questions_builder to avoid unnecessary if ... else
After trying several solutions I like this one, it seems evolutive & flexible.

Most of the methods will end in FieldsBuilder by delegation.

What do you think?

### CODE REVIEW

File: app/services/importer/greenhouse_fields_builder.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# Each section includes details like title, description, build type (API in this case), and individual questions.
+# Question details incorporate attributes, labels, descriptions, required status, and associated field objects.
+# The output of this service is used to the job associated application_question_set.
+
+module Importer
+  class GreenhouseFieldsBuilder < ApplicationTask
+    include FaradayHelpers
+
+    def initialize(data)
+      @data = Importer::GreenhouseFieldsFormatter.call(data.with_indifferent_access)
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_builder.rb]: Is it acceptable to call another service object in the initialise method?

File: app/services/importer/greenhouse_fields_builder.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    end
+
+    def questions_builder(section)
+      return [] unless section.present?
+
+      section.map { |question| question_params(question) if question.present? }
+    end
+
+    def question_params(question)
+      { attribute: question[:fields].first[:name],
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_builder.rb]: Update formatting?
{
...
}

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    def select_transform_data
+      {
+        core_questions: { title: nil, description: nil, questions: @data[:questions] },
+        demographic_questions: demographic_formatter(@data.dig(:demographic_questions, :questions)),
+        compliance_questions: compliance_formatter(@data[:compliance]),
+        location_questions: { title: nil, description: nil, questions: location_formatter(@data[:location_questions]) }
+      }
+    end
+
+    ###
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb:24]: Remove?

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+      {
+        title: section_attribute_finder(:compliance, :title) || "EEOC compliance questions",
+        description: section_attribute_finder(:compliance, :description),
+        questions: compliance_questions(compliance_data)
+      }
+    end
+
+    def compliance_questions(compliance_data)
+      compliance_data.map do |section|
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb:37]: Interesting that we have all these nested maps - do we need to be conscious of how this performs?

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    def demographic_questions(demographic_data)
+      demographic_data.map do |question|
+        attribute = question[:label].parameterize.underscore.first(50)
+        values = question[:answer_options].map { |option| option.transform_keys({ 'id' => 'value' }) }
+        fields = [{ name: question[:id], type: question[:type], values: }]
+        { attribute:, description: nil, label: question[:label], required: question[:required], fields: }
+      end
+    end
+
+    def location_formatter(location_data)
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb:77]: Is a separate location formatter required?

File: app/models/concerns/ats/greenhouse/application_fields.rb [hunk 6]
```diff
@@ -1,8 +1,9 @@
 module Ats
   module Greenhouse
     module ApplicationFields
-      def get_application_question_set(job, data)
-        Importer::GetGreenhouseFields.call(job, data)
+      def get_application_question_set(_job, data)
```
Comment (Ches-ctrl) [inline app/models/concerns/ats/greenhouse/application_fields.rb:4]: I'm wondering if we need to go to the modules anymore - presumably we can just call the service object directly if we know the ATS? One to discuss as the advantage of the current setup is that we already know which ATS this calls so wouldn't need to route this otherwise - it may not be an improvement to change the setup.

File: app/services/importer/greenhouse_fields_builder.rb [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      build_fields
+      log_and_return_fields
+    end
+
+    ###
+
+    def build_fields
+      @fields = [
+        {
+          build_type: :api,
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_builder.rb]: We don't need the build_type for each hash presumably?

File: app/services/importer/greenhouse_fields_builder.rb [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+          section_slug: "core_fields",
+          title: "Main application",
+          description: nil,
+          questions: questions_builder(@data.dig(:core_questions, :questions))
+        },
+        {
+          build_type: :api,
+          section_slug: "demographic_fields",
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_builder.rb]: Do we think section slug should be built dynamically? Presumably also we don't need to write out all 3 hashes as we can just loop through the returned data now that the method for building the response is the same? This solution is starting to look quite neat

File: app/services/importer/greenhouse_fields_builder.rb [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+          description: @data.dig(:demographic_questions, :description),
+          questions: questions_builder(@data.dig(:demographic_questions, :questions))
+        },
+        {
+          build_type: :api,
+          section_slug: "compliance_fields",
+          title: @data.dig(:compliance_questions, :title),
+          description: @data.dig(:compliance_questions, :description),
+          questions: questions_builder(@data.dig(:compliance_questions, :questions))
+        }
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_builder.rb]: Are there any other types of sections?

File: app/services/importer/greenhouse_fields_builder.rb [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+      section.map { |question| question_params(question) if question.present? }
+    end
+
+    def question_params(question)
+      { attribute: question[:fields].first[:name],
+        description: question[:description],
+        label: question[:label],
+        required: question[:required],
+        fields: question[:fields] }
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_builder.rb]: The value of standardising the structure of ApplicationQuestionSet is clear here

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+      @data
+    end
+
+    def process
+      select_transform_data
+    end
+
+    def select_transform_data
+      {
+        core_questions: { title: nil, description: nil, questions: @data[:questions] },
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb]: So we standardise each type of question_set according to the core_questions structure

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    end
+
+    ###
+
+    def compliance_formatter(compliance_data)
+      return {} unless compliance_data
+
+      {
+        title: section_attribute_finder(:compliance, :title) || "EEOC compliance questions",
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb]: Let's align on the title

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+      {
+        title: section_attribute_finder(:demographic_questions, :header),
+        description: section_attribute_finder(:demographic_questions, :description),
+        questions: demographic_questions(demographic_data)
+      }
+    end
+
+    def demographic_questions(demographic_data)
+      demographic_data.map do |question|
+        attribute = question[:label].parameterize.underscore.first(50)
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb]: I think we need to remove attributes as a thing - to discuss

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+        values = question[:answer_options].map { |option| option.transform_keys({ 'id' => 'value' }) }
+        fields = [{ name: question[:id], type: question[:type], values: }]
+        { attribute:, description: nil, label: question[:label], required: question[:required], fields: }
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb]: Lets align on the best ordering of the hash

**Review by Ches-ctrl — COMMENTED** [review]
This should be plug and play now yes? Are there any outstanding issues? Also why is the test failing? Won't load for me

File: app/services/importer/greenhouse_fields_builder.rb (hunk 2 above)
Comment (patanj101) [reply app/services/importer/greenhouse_fields_builder.rb]: done

File: app/services/importer/greenhouse_fields_formatter.rb (hunk 13 above)
Comment (patanj101) [reply app/services/importer/greenhouse_fields_formatter.rb]: done

File: app/services/importer/greenhouse_fields_formatter.rb (hunk 14 above)
Comment (patanj101) [reply app/services/importer/greenhouse_fields_formatter.rb]: You can choose and tell me

File: app/services/importer/greenhouse_fields_builder.rb (hunk 9 above)
Comment (patanj101) [reply app/services/importer/greenhouse_fields_builder.rb]: Such a good point you had this morning.
Found the solution to manage all types of sections

File: spec/models/applicant_tracking_system_spec.rb [hunk 15]
```diff
@@ -188,7 +188,7 @@
         title = feed.dig('jobs', 0, 'title')
         job_id = feed.dig('jobs', 0, 'id')
         company = CompanyCreator.call(ats: @ashbyhq, ats_identifier: 'lightdash')
-        job = JobCreator.call(ats: @ashbyhq, company:, job_id:)
+        job = JobCreator.call(ats: @ashbyhq, company:, job_id:, data: feed)
```
Comment (Ches-ctrl) [inline spec/models/applicant_tracking_system_spec.rb]: Why do we have a data: feed now?

File: app/services/importer/ashbyhq_fields_formatter.rb [hunk 16]
```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module Importer
+  # This service object is responsible for formatting data received from Ashbyhq, another recruiting platform, into a structured format suitable for our application.
+  # It extracts specific sections (core questions and surveys) and transforms them into a standardized format.
```
Comment (Ches-ctrl) [inline app/services/importer/ashbyhq_fields_formatter.rb:5]: What is the standardised format that it outputs it in? Can you show me an example?

The point we were discussing with Dan yesterday is why do we standardise the format twice, once in the formatter and again in the field builder? 

Wouldn't it make sense to just format it once?

File: app/services/importer/ashbyhq_fields_formatter.rb [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+    def survey_questions(survey_data)
+      survey_data.map do |question|
+        attribute = question[:label].parameterize.underscore.first(50)
+        values = question[:answer_options].map { |option| option.transform_keys({ 'id' => 'value' }) }
+        fields = [{ name: question[:id], type: question[:type], values: }]
+        { attribute:, description: nil, label: question[:label], required: question[:required], fields: }
+      end
+    end
+
+    INPUT_TYPES = {
```
Comment (Ches-ctrl) [inline app/services/importer/ashbyhq_fields_formatter.rb:103]: Wondering if we want this to sit separately somewhere

File: app/services/importer/fields_builder.rb [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+      build_fields
+      log_and_return_fields
+    end
+
+    ###
+
+    def build_fields
+      @fields = @data.map do |section_sym, section_data|
+        {
+          build_type:,
```
Comment (Ches-ctrl) [inline app/services/importer/fields_builder.rb]: To remove build_type for each section once converted to array

File: app/services/importer/greenhouse_fields_formatter.rb (hunk 13 above)
Comment (Ches-ctrl) [reply app/services/importer/greenhouse_fields_formatter.rb]: Need to check with Dan on this as we sometimes get this from the ATS e.g. Greenhouse gives us full_name - need to make sure we're still capturing that info

File: app/models/concerns/ats/ashbyhq/application_fields.rb [hunk 19]
```diff
@@ -4,18 +4,18 @@ module ApplicationFields
...
+        job.update(deadline: data['applicationDeadline'])
+        formatted_data = Importer::AshbyhqFieldsFormatter.call(job, data.with_indifferent_access)
+        Importer::FieldsBuilder.call(formatted_data)
       end
 
       private
 
-      def fetch_job_api_data(job_id, ats_identifier)
-        faraday_request(request_details(job_id, ats_identifier))
+      def fetch_job_api_data(job)
```
Comment (Ches-ctrl) [inline app/models/concerns/ats/ashbyhq/application_fields.rb:17]: Why pass the job instead?

File: app/models/concerns/ats/ashbyhq/application_fields.rb [hunk 20]
```diff
@@ -4,18 +4,18 @@ module ApplicationFields
...
       def get_application_question_set(job, _data)
-        p "Getting AshbyHQ application criteria"
-        data = fetch_job_api_data(job.ats_job_id, job.company.ats_identifier)
-        return [] unless (job_data = data&.dig('data', 'jobPosting'))
+        data = fetch_job_api_data(job)
+        return [] unless (data = data&.dig('data', 'jobPosting'))
 
-        job.update(deadline: job_data['applicationDeadline']) # not sure what format
-        Importer::GetAshbyFields.call(job, job_data)
+        job.update(deadline: data['applicationDeadline'])
```
Comment (Ches-ctrl) [inline app/models/concerns/ats/ashbyhq/application_fields.rb:10]: Think we can standardise this a bit further between ATS systems - to discuss

File: app/models/concerns/ats/ashbyhq/xxx_application_fields.rb [hunk 21]
```diff
@@ -0,0 +1,43 @@
+module Ats
```
Comment (Ches-ctrl) [inline app/models/concerns/ats/ashbyhq/xxx_application_fields.rb:1]: This is a template class presumably?

**Review by Ches-ctrl — COMMENTED** [review]
2nd round of review

File: app/models/concerns/ats/ashbyhq/xxx_application_fields.rb (hunk 21 above)
Comment (patanj101) [reply app/models/concerns/ats/ashbyhq/xxx_application_fields.rb:1]: I will delete it. It was a easy fix to have a backup of the previous version of that file.

File: app/models/concerns/ats/ashbyhq/application_fields.rb (hunk 20 above)
Comment (patanj101) [reply app/models/concerns/ats/ashbyhq/application_fields.rb:10]: I don't like that part but did not have enough knowledge to decide what to do with.
to be discussed with Dan.

File: app/models/concerns/ats/ashbyhq/application_fields.rb (hunk 19 above)
Comment (patanj101) [reply app/models/concerns/ats/ashbyhq/application_fields.rb:17]: Small refactoring ... i prefer to send only one parameter than 2 coming from the same object (job in that case)

File: app/services/importer/ashbyhq_fields_formatter.rb (hunk 17 above)
Comment (patanj101) [reply app/services/importer/ashbyhq_fields_formatter.rb:103]: it is totally dependent to ashby it can stay here for the moment ... maybe later we can add a input_types_ats configuration file

File: app/services/importer/ashbyhq_fields_formatter.rb (hunk 16 above)
Comment (patanj101) [reply app/services/importer/ashbyhq_fields_formatter.rb:5]: one format, the other build.
it is a separation of concern.
if you ask me to transform application_question_set from array to hash ... at the moment i will have to change only one file.
If we go with your proposition i will have to change every ats_formatter.

File: spec/models/applicant_tracking_system_spec.rb (hunk 15 above)
Comment (patanj101) [reply spec/models/applicant_tracking_system_spec.rb]: I think I need to refactor that part at some point.
It seems useless if you dive into the JobCreator but if we don't need data (because we are going to fetch it later) why do we create feed in the first place?

**Comment by patanj101** [comment]
Can you check my last comments

File: app/services/importer/ashbyhq_fields_formatter.rb (hunk 16 above)
Comment (Ches-ctrl) [reply app/services/importer/ashbyhq_fields_formatter.rb:5]: Can you give an example of what the formatter outputs?

File: spec/models/applicant_tracking_system_spec.rb (hunk 15 above)
Comment (Ches-ctrl) [reply spec/models/applicant_tracking_system_spec.rb]: Sorry I'm confused - didn't you add it? What is it doing?

Comment (patanj101) [reply spec/models/applicant_tracking_system_spec.rb]: I added it. But i did because of confusion. I remove it

File: app/models/application_question_set.rb [hunk 22]
```diff
@@ -18,12 +18,14 @@ def question_by_attribute(attribute)
...
+    form_structure.map do |_section, section_data|
+      next unless section_data.is_a?(Hash)
+
+      section_data['questions'].map { |question| ApplicationQuestion.new(question.merge(section: section_data['title'])) }
+    end.compact.flatten
   end
 
   def no_of_qs
-    form_structure.map { |section| section[:questions] }.flatten.count
+    form_structure.select { |section| section['questions'] }.flatten.count
```
Comment (Ches-ctrl) [inline app/models/application_question_set.rb:29]: why the change from symbols to strings?

File: app/models/concerns/ats/ashbyhq/application_fields.rb (hunk 20 above)
Comment (Ches-ctrl) [reply app/models/concerns/ats/ashbyhq/application_fields.rb:10]: @daniel-sussman - one to discuss

File: app/services/importer/fields_builder.rb [hunk 23]
```diff
@@ -0,0 +1,71 @@
...
+      @fields
+    end
+
+    def questions_builder(section)
+      return [] unless section.present?
+
+      section.map { |question| question_params(question) if question.present? }
+    end
+
+    def question_params(question)
```
Comment (Ches-ctrl) [inline app/services/importer/fields_builder.rb:61]: Can we store an example of a built AQS somewhere and also have this be able to print out a build

File: app/services/importer/greenhouse_fields_formatter.rb (hunk 5 above)
Comment (Ches-ctrl) [reply app/services/importer/greenhouse_fields_formatter.rb:77]: Did you come to a conclusion on this?

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 24]
```diff
@@ -0,0 +1,157 @@
...
+
+    def attribute_strict_match(key)
+      ATTRIBUTES_DICTIONNARY[key]
+    end
+
+    def default_attribute(question)
+      question[:label].parameterize.underscore.first(60)
+    end
+
+    ATTRIBUTES_DICTIONNARY = {
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb]: Dictionary has an extra N - please remove

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 25]
```diff
@@ -0,0 +1,157 @@
...
+      'email' => 'email',
+      'location' => 'location',
+      'resume' => 'resume',
+      'phone' => 'phone_number',
+      'linkedin' => 'linkedin',
+      'cover_letter' => 'cover_letter'
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb:124]: Is this the complete list?

File: app/services/importer/greenhouse_fields_formatter.rb [hunk 26]
```diff
@@ -0,0 +1,157 @@
...
+    ###
+    ### types
+    ###
+
+    INPUT_TYPES = {
+      'input_file' => :upload,
+      'input_text' => :input,
+      'multi_value_multi_select' => :multi_select,
+      'multi_value_single_select' => :select,
+      'textarea' => :textarea
```
Comment (Ches-ctrl) [inline app/services/importer/greenhouse_fields_formatter.rb:136]: Similarly complete?

File: app/tasks/job_attributes_fetcher.rb [hunk 27]
```diff
@@ -39,9 +39,9 @@ def application_question_set
   end
 
   def apply_with_cheddar
-    return false unless application_question_set&.first&.dig(:questions)
+    return false unless application_question_set.dig(:core_questions, :questions)
 
-    application_question_set.first[:questions]
+    application_question_set.dig(:core_questions, :questions)
                             .none? { |question| apply_with_cheddar_conditions(question) }
```
Comment (Ches-ctrl) [inline app/tasks/job_attributes_fetcher.rb]: Can we add comments to this file please for what it does

Comment (Ches-ctrl) [reply app/tasks/job_attributes_fetcher.rb]: Also do we need that many variables in the initialise method?

File: app/tasks/last_applicant_answers_fetcher.rb [hunk 28]
```diff
@@ -4,7 +4,7 @@ class LastApplicantAnswersFetcher < ApplicationTask
   def initialize(current_user)
     @current_user = current_user
     @user_detail = current_user.user_detail
-    @job_applications = current_user.job_applications.where(status: "completed")
+    @job_applications = current_user.job_applications # .where(status: "completed")
```
Comment (Ches-ctrl) [inline app/tasks/last_applicant_answers_fetcher.rb:11]: Why has this changed? What is its impact?

File: app/tasks/last_applicant_answers_fetcher.rb [hunk 29]
```diff
@@ -4,7 +4,7 @@ class LastApplicantAnswersFetcher < ApplicationTask
   def initialize(current_user)
     @current_user = current_user
     @user_detail = current_user.user_detail
-    @job_applications = current_user.job_applications.where(status: "completed")
```
Comment (Ches-ctrl) [inline app/tasks/last_applicant_answers_fetcher.rb:7]: '+ Comments to this file please

**Review by Ches-ctrl — CHANGES_REQUESTED** [review]
3rd Review

File: app/services/importer/fields_builder.rb (hunk 23 above)
Comment (patanj101) [reply app/services/importer/fields_builder.rb:61]: The output of fields_builder is applicaton_question_set.form_structure
You can have it for any job through your console.
Through : job.application_question.set.form_structure.

Would you like something else ? Except a comment with the structure?

File: app/services/importer/greenhouse_fields_formatter.rb (hunk 5 above)
Comment (patanj101) [reply app/services/importer/greenhouse_fields_formatter.rb:77]: Your question doesn't make sense to me ...
It is a different section so i need a separate formatter ... it might change in the future if i see a pattern between ATS but i don't at the moment.

File: app/services/importer/greenhouse_fields_formatter.rb (hunk 25 above)
Comment (patanj101) [reply app/services/importer/greenhouse_fields_formatter.rb:124]: So far yes ... took it from Dan.

File: app/services/importer/greenhouse_fields_formatter.rb (hunk 26 above)
Comment (patanj101) [reply app/services/importer/greenhouse_fields_formatter.rb:136]: So far yes ... took it from Dan.

File: app/tasks/job_attributes_fetcher.rb (hunk 27 above)
Comment (patanj101) [reply app/tasks/job_attributes_fetcher.rb]: I did  not work much on that file. ... I don't know.

File: app/tasks/last_applicant_answers_fetcher.rb (hunk 28 above)
Comment (patanj101) [reply app/tasks/last_applicant_answers_fetcher.rb:11]: Temporary solution to be able to test pre-filler and see the gap between array version to hash version.
Hash broke many things and most of the application_forms are not going to the state completed. So i remove it to be able to have more information to be able to debug.

File: app/models/application_question_set.rb (hunk 22 above)
Comment (patanj101) [reply app/models/application_question_set.rb:29]: After being saved as a jsonb in database section[:questions] is not accessible anymore.

Screenshot from 2024-08-06 20-32-08

File: app/services/importer/get_bamboo_fields.rb [hunk 30]
```diff
@@ -0,0 +1,319 @@
+# frozen_string_literal: true
+
+module Importer
+  class GetBambooFields < GetApiFields
```
Comment (Ches-ctrl) [inline app/services/importer/get_bamboo_fields.rb]: Just confirming this is only included as the old version - expecting something to look different here

File: public/ashby_hq_aqs_builder_output.json [hunk 31]
```diff
@@ -0,0 +1 @@
...null,"label":"Name","required":true,"fields":[{"name":"_systemfield_name","selector":null,"type":"input","options":[]}]},{"attribute":"email","description":null,"label":"Email","required":true,"fields":[{"name":"_systemfield_email","selector":null,"type":"input","options":[]}]},{"attribute":"linkedin","description":null,"label":"LinkedIn","required":true,"fields":[{"name":"05acaa26-5e32-496a-be01-6aae72db3e27","selector":null,"type":"input","options":[]}]},{"attribute":"would_you_require_a_visa_sponsorship_to_relocate_to_london","description":null,"label":"Would you require a visa sponsorship to relocate to London?","required":true,"fields":[{"name":"00b30aa4-0b2a-4e60-b5b7-c80e29d938e2","selector":null,"type":"boolean","options":[]}]},{"attribute":"resume","description":null,"label":"Resume","required":true,"fields":[{"name":"_systemfield_resume","selector":null,"type":"upload","options":[]}]},{"attribute":"why_do_you_want_to_join_us","description":null,"label":"Why do you want to join us?","required":true,"fields":[{"name":"0e089a26-b553-494b-aede-1c03030b6285","selector":null,"type":"textarea","options":[]}]}]},"survey_questions":{"title":null,"description":null,"questions":[]}}
```
Comment (Ches-ctrl) [inline public/ashby_hq_aqs_builder_output.json]: Can you use pretty_generate for all of these JSON files please going forward. Please notify me once done and I will review. Can we also having these all sit in a folder within public so they're together please. /public/jsons/XXX

**Review by Ches-ctrl — DISMISSED** [review]
4th Review

File: public/ashby_hq_aqs_builder_output.json (hunk 31 above)
Comment (Ches-ctrl) [reply public/ashby_hq_aqs_builder_output.json]: @patanj101 - one addition to this morning - can you update this to that it's in the output I'm expecting

i.e.:

{
 XXX
}

**Review by daniel-sussman — APPROVED** [review]

File: public/ashby_hq_aqs_builder_output.json (hunk 31 above)
Comment (patanj101) [reply public/ashby_hq_aqs_builder_output.json]: old files ... i just removed them
